### PR TITLE
DOM manipulation removal

### PR DESCRIPTION
**Pull request by willespencer**
### Summary 

Remove most of the existing bad DOM Manipulation code in the codebase and replace it with event handling per issue #10. This will make it much easier in the future to see and change how data flows between components.

- [x] Removed almost all instances of `document.getElementByID` or `.innerHTML` that grabs and manipulates data poorly, especially in opening/closing modals
- [x] Removed almost all `$parent` references.
- [x] Removed other outdated/duplicated code and comments (2 `<semester>`s, NewCustomCourse.vue, etc.)
- [ ] Add code that allows you to click outside of modals to close them without DOM Manipulation ([see Monday](https://courseplan.monday.com/boards/433956786/pulses/848083188))
- [ ] Refactor the add course modal to avoid using DOM Manipulation in `Modal.vue`, `NewCourse.vue`, and `OnboardingTransfer.vue` (waiting for @hahnbeelee's work to be finished before messing around with this file, will take significant more work than these other elements, [see Monday](https://courseplan.monday.com/boards/433956786/pulses/848072269))
- [ ] Refactor the direction of the triple dot menu on semesters (still need a better solution for this, [see Monday](https://courseplan.monday.com/boards/433956786/pulses/542946252))
- [ ] Refactor `createCourse` so it is not called with `$parent` (@SamChou19815 do you want to take this?)

### Test Plan 

Ensure that everything works on the site compared to master (https://cornelldti-courseplan-dev.web.app/) with the exception of clicking outside of modals closing them.

Check if all of the following modals can be opened, closed, and used as expected
- Add Course
- New Semester
- Edit Semester
- Delete Semester

### Notes 

The rest of the DOM Manipulation should be removed so it is not duplicated in the future like it has been since CoursePlan started. 

### Breaking Changes  

As specified above, just clicking outside of modals.

### CODE REVIEW

**Comment by SamChou19815** [comment]
@willespencer when do you think it will be ready? I might want to touch some of the files soon.

**Review by willespencer — COMMENTED** [review]
@SamChou19815 I'll get it ready for review by the dev meeting!

File: src/components/Modals/Modal.vue [hunk 1]
```diff
@@ -26,27 +28,46 @@
...
+        vnode.context[binding.expression](event);
+      }
+    };
+    document.body.addEventListener('click', el.event);
+  },
+  unbind(el) {
+    document.body.removeEventListener('click', el.event);
+  }
+};
+
```
Comment (SamChou19815) [inline src/components/Modals/Modal.vue:51]: no copy-pasta. see #187

**Comment by SamChou19815** [comment]
@willespencer I think it's better for this PR to stop here. It's already start to get big and is blocking some other refactors I plan to do.

**Review by willespencer — COMMENTED** [review]
@SamChou19815 Agreed! Just making sure I document the remaining issues and things I have figured tried/figured out in this PR and on Monday, and then I'll mark it ready to review 😄.

File: package-lock.json [hunk 2]
```diff
@@ -10253,8 +10253,7 @@
         "ansi-regex": {
```
Comment (SamChou19815) [inline package-lock.json:10253]: why do we have a package-lock change here?

**Review by SamChou19815 — APPROVED** [review]
tested the features and they all seem to work

**Pull request merged**